cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerLoggingConfigurationDiff.java

```diff
@@ -0,0 +1,186 @@
+        Map<String, String> desiredMap = orderedProperties.asMap();
+
+        JsonNode source = patchMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true).valueToTree(currentMap);
+        JsonNode target = patchMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true).valueToTree(desiredMap);
+        JsonNode jsonDiff = JsonDiff.asJson(source, target);
+
+        for (JsonNode d : jsonDiff) {
+            String pathValue = d.get("path").asText();
+            String pathValueWithoutSlash = pathValue.substring(1);
+
+            Optional<ConfigEntry> optEntry = brokerConfigs.entries().stream()
+                    .filter(configEntry -> configEntry.name().equals(pathValueWithoutSlash))
+                    .findFirst();
+
+            if (pathValueWithoutSlash.equals("log4j.rootLogger")) {
+                if (!desiredMap.get(pathValueWithoutSlash).matches(".+,.+")) {
+                    log.warn("Broker {} logging: Logger log4j.rootLogger should contain level and appender, e.g. \'log4j.rootLogger = INFO, CONSOLE\'", brokerId);
+                }
+            }
+            String op = d.get("op").asText();
+            if (optEntry.isPresent()) {
+                ConfigEntry entry = optEntry.get();
+                if ("remove".equals(op)) {
+                    removeProperty(updatedCE, pathValueWithoutSlash, entry);
+                } else if ("replace".equals(op)) {
+                    // entry is in the current, desired is updated value
+                    if (!entry.value().equals(parseLogLevelFromAppenderCouple(desiredMap.get(entry.name())))) {
+                        updateOrAdd(entry.name(), desiredMap, updatedCE);
+                    }
+                }
+            } else {
+                if ("add".equals(op)) {
+                    if (pathValueWithoutSlash.equals("log4j.rootLogger")) {
+                        String level = parseLogLevelFromAppenderCouple(desiredMap.get(pathValueWithoutSlash));
+                        if (!level.equals(currentMap.get("root"))) {
+                            updateOrAddRoot(level, updatedCE);
+                        }
+                    } else {
+                        // entry is not in the current, it is added
+                        updateOrAdd(pathValueWithoutSlash, desiredMap, updatedCE);
+                    }
+                }
+            }
+
+            log.debug("Kafka Broker {} Logging Config Differs : {}", brokerId, d);
+            log.debug("Current Kafka Broker Logging Config path {} has value {}", pathValueWithoutSlash, lookupPath(source, pathValue));
+            log.debug("Desired Kafka Broker Logging Config path {} has value {}", pathValueWithoutSlash, lookupPath(target, pathValue));
+        }
+        return updatedCE;
+    }
+
+    private static String parseLogLevelFromAppenderCouple(String lev) {
+        String[] arr = lev.split(",");
+        String level = arr[0].replaceAll("\\s", "");
+        return level;
```

Comment:
More efficient to use `indexOf(",")` + `trim()`. Both `split` and `replaceAll` will have to use `Pattern` under the hood.